src/index/mod.rs

```diff
@@ -113,30 +114,48 @@ impl<R: BufRead + Seek> IndexReader for Index<R> {
         // Normalize query according to the metadata
         let mut headword = headword.to_string();
         normalize_headword(&mut headword, &self.metadata);
-        let headword: &str = &headword;
+        let headword: &str = headword.trim();
 
         if fuzzy {
             let results: Vec<Entry> = self.entries
                 .iter()
-                .filter(|entry| levenshtein(headword, &entry.headword) <= 1)
+                .filter(|entry| {
+                    if relaxed {
+                        let transliterated = unidecode(&entry.headword);
+                        levenshtein(headword, transliterated.trim()) <= 1
+                    } else {
+                        levenshtein(headword, &entry.headword) <= 1
+                    }
+                })
                 .cloned()
                 .collect();
 
             if results.is_empty() { return Err(WordNotFound(headword.into())) }
 
             Ok(results)
-        } else if let Ok(pivot) = self.entries.binary_search_by_key(&headword, |entry| &entry.headword) {
+        } else if let Ok(pivot) = self.entries.binary_search_by(|entry| {
+                if relaxed {
+                    let transliterated = unidecode(&entry.headword);
+                    transliterated.trim().cmp(headword)
```

Comment:
This isn't going to work: `entries` is sorted by `entry.headword`, not by `unidecode(&entry.headword)`.

Here's an excerpt from `binary_search_by`'s documentation that underlines my point:

> the comparator function should implement an order consistent with the sort order of the underlying slice

Reply:
Yep, good catch, thanks

Reply:
Should be fixed now